tests/integration/test_agentd/test_agentd_state/test_agentd_state.py

```diff
@@ -0,0 +1,382 @@
+'''
+copyright: Copyright (C) 2015-2022, Wazuh Inc.
+
+           Created by Wazuh, Inc. <info@wazuh.com>.
+
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+type: integration
+
+brief: The 'wazuh-agentd' program is the client-side daemon that communicates with the server.
+       These tests will check if the content of the 'wazuh-agentd' daemon statistics file is valid.
+       The statistics files are documents that show real-time information about the Wazuh environment.
+
+components:
+    - agentd
+
+targets:
+    - agent
+
+daemons:
+    - wazuh-agentd
+    - wazuh-remoted
+
+os_platform:
+    - linux
+    - windows
+
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - Debian Buster
+    - Red Hat 8
+    - Ubuntu Focal
+    - Ubuntu Bionic
+    - Windows 10
+    - Windows Server 2019
+    - Windows Server 2016
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/reference/statistics-files/wazuh-agentd-state.html
+
+tags:
+    - stats_file
+'''
+
+import json
+import pytest
+from pathlib import Path
+import sys
+import time
+
+from wazuh_testing.constants.paths.variables import AGENTD_STATE
+from wazuh_testing.constants.platforms import WINDOWS
+from wazuh_testing.modules.agentd.configuration import AGENTD_DEBUG, AGENTD_WINDOWS_DEBUG
+from wazuh_testing.tools.simulators.remoted_simulator import RemotedSimulator
+from wazuh_testing.utils.configuration import get_test_cases_data, load_configuration_template
+
+from . import CONFIGS_PATH, TEST_CASES_PATH
+from utils import wait_keepalive, wait_ack, wait_state_update, wait_agent_notification
+
+# Marks
+pytestmark = pytest.mark.tier(level=0)
+
+# Configuration and cases data.
+configs_path = Path(CONFIGS_PATH, 'wazuh_conf.yaml')
+cases_path = Path(TEST_CASES_PATH, 'wazuh_state_config_tests.yaml')
+
+# Test configurations.
+config_parameters, test_metadata, test_cases_ids = get_test_cases_data(cases_path)
+test_configuration = load_configuration_template(configs_path, config_parameters, test_metadata)
+
+if sys.platform == WINDOWS:
+    local_internal_options = {AGENTD_WINDOWS_DEBUG: '2'}
+else:
+    local_internal_options = {AGENTD_DEBUG: '2'}
+
+daemons_handler_configuration = {'all_daemons': True}
+
+def start_remoted_server(test_metadata) -> None:
+    """"Start RemotedSimulator if test case need it"""
+    if 'remoted' in test_metadata and test_metadata['remoted']:
+        remoted_server = RemotedSimulator()
+        remoted_server.start()
+    else:
+        remoted_server = None
+    return remoted_server
+
```

Comment:
Remember to always leave two empty lines between functions. Check all files.

Reply:
Addressed with https://github.com/wazuh/wazuh/pull/18353/commits/5a1b7778d254f00b85c507ace508492c18df7d0c